File: lib/src/notification_category.dart
import 'package:flutter_local_notifications/src/notification_action.dart';

class NotificationCategory {
  final String identifier;

  final List<NotificationAction> actions;

  NotificationCategory(this.identifier, this.actions);

  Map<String, Object> toMap() => {
        'identifier': identifier,
        'actions': actions.map((a) => a.toMap()).toList(),
      };
}

File: android/src/main/java/com/dexterous/flutterlocalnotifications/models/NotificationCategory.java
package com.dexterous.flutterlocalnotifications.models;

import java.util.List;

public class NotificationCategory {
    private NotificationCategory(String identifier, List<NotificationAction> actions) {
        this.identifier = identifier;
        this.actions = actions;
    }

    public String identifier;

    public List<NotificationAction> actions;
}
